common/test/acceptance/tests/helpers.py

```diff
@@ -313,13 +313,18 @@ def get_matching_events(self, event_type):
             "time": {"$gt": self.start_time},
         })
 
-    def verify_events_of_type(self, event_type, expected_events):
+    def verify_events_of_type(self, event_type, expected_events, expected_referers=None):
         """Verify that the expected events of a given type were logged.
 
         Args:
             event_type (str): The type of event to be verified.
             expected_events (list): A list of dicts representing the events that should
                 have been fired.
+            expected_referers (list): A list of strings representing the referers for each event
+                that should been fired (optional). If present, the actual referers compared
+                with this list, checking that the expected_referers are the suffixes of
+                actual_referers. For example, if one event is expected, specifying ["/account/settings"]
+                will verify that the referer for the single event ends with "/account/settings".
```

Comment:
Interesting idea. I'll think about it, but not hold up this PR right now (we have a stack of PRs going, and some of them are already using this method as-is).